experimental/dds/tree2/src/shared-tree/sharedTreeChangeCodecs.ts

```diff
@@ -0,0 +1,78 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+	IJsonCodec,
+	ICodecOptions,
+	ICodecFamily,
+	makeCodecFamily,
+	DiscriminatedUnionDispatcher,
+} from "../codec";
+import { Mutable } from "../util";
+import {
+	EncodedModularChangeset,
+	ModularChangeset,
+	makeSchemaChangeCodec,
+} from "../feature-libraries";
+import { SharedTreeChange, SharedTreeInnerChange } from "./sharedTreeChangeTypes";
+import { EncodedSharedTreeChange, EncodedSharedTreeInnerChange } from "./sharedTreeChangeFormat";
+
+export function makeSharedTreeChangeCodec(
+	modularChangeCodec: IJsonCodec<ModularChangeset, EncodedModularChangeset>,
+	{ jsonValidator: validator }: ICodecOptions,
+): IJsonCodec<SharedTreeChange, EncodedSharedTreeChange> {
+	const schemaChangeCodec = makeSchemaChangeCodec({ jsonValidator: validator });
```

Comment:
this looses any possible codec options other than the validator. I think it would be better to not destructure and recreate the options like this.

Reply:
Done.